libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo/gradle/api/Service.kt

```diff
@@ -135,41 +135,104 @@ interface Service {
+   */
+  @ApolloExperimental
+  fun mapScalar(graphQLName: String, targetName: String, expression: String, inlineClassProperty: String? = null)
+
   /**
    * Map the given GraphQL scalar to [kotlin.String] and use the builtin adapter
    */
   fun mapScalarToKotlinString(graphQLName: String)
 
+  /**
+   * Map the given GraphQL scalar to a [kotlin.String] wrapped in an inline class and use the builtin adapter
+   */
+  @ApolloExperimental
+  fun mapScalarToKotlinString(graphQLName: String, inlineClass: String, inlineClassProperty: String)
+
   /**
    * Map the given GraphQL scalar to [kotlin.Int] and use the builtin adapter
    */
   fun mapScalarToKotlinInt(graphQLName: String)
 
+  /**
+   * Map the given GraphQL scalar to a [kotlin.Int] wrapped in an inline class and use the builtin adapter
+   */
+  @ApolloExperimental
+  fun mapScalarToKotlinInt(graphQLName: String, inlineClass: String, inlineClassProperty: String)
+
   /**
    * Map the given GraphQL scalar to [kotlin.Double] and use the builtin adapter
    */
   fun mapScalarToKotlinDouble(graphQLName: String)
 
+  /**
+   * Map the given GraphQL scalar to a [kotlin.Double] wrapped in an inline class and use the builtin adapter
+   */
+  @ApolloExperimental
+  fun mapScalarToKotlinDouble(graphQLName: String, inlineClass: String, inlineClassProperty: String)
+
   /**
    * Map the given GraphQL scalar to [kotlin.Float] and use the builtin adapter
    */
   fun mapScalarToKotlinFloat(graphQLName: String)
 
+  /**
+   * Map the given GraphQL scalar to a [kotlin.Float] wrapped in an inline class and use the builtin adapter
+   */
+  @ApolloExperimental
+  fun mapScalarToKotlinFloat(graphQLName: String, inlineClass: String, inlineClassProperty: String)
+
   /**
    * Map the given GraphQL scalar to [kotlin.Long] and use the builtin adapter
    */
   fun mapScalarToKotlinLong(graphQLName: String)
 
+  /**
+   * Map the given GraphQL scalar to a [kotlin.Long] wrapped in an inline class and use the builtin adapter
+   */
+  @ApolloExperimental
+  fun mapScalarToKotlinLong(graphQLName: String, inlineClass: String, inlineClassProperty: String)
+
   /**
    * Map the given GraphQL scalar to [kotlin.Boolean] and use the builtin adapter
    */
   fun mapScalarToKotlinBoolean(graphQLName: String)
 
+  /**
+   * Map the given GraphQL scalar to a [kotlin.Boolean] wrapped in an inline class and use the builtin adapter
+   */
+  @ApolloExperimental
+  fun mapScalarToKotlinBoolean(graphQLName: String, inlineClass: String, inlineClassProperty: String)
+
   /**
    * Map the given GraphQL scalar to [kotlin.Any] and use the builtin adapter
    */
   fun mapScalarToKotlinAny(graphQLName: String)
 
+  /**
+   * Map the given GraphQL scalar to a [kotlin.Any] wrapped in an inline class and use the builtin adapter
+   */
+  @ApolloExperimental
+  fun mapScalarToKotlinAny(graphQLName: String, inlineClass: String, inlineClassProperty: String)
+
```

Comment:
Can you remove the overloads? 

* They sound a bit weird to me as it's not mapping to String/Boolean/etc... but to a generated inline class that as a String/Boolean/etc... property
* They create a secondary source of truth: what happens if someone uses that + `@inlineClass` ? Who takes precedence? [Precedence is bad](https://youtu.be/YZstpc2939s?t=1965).
* Let's preserve option value? We can always add them later if needed?

Reply:
Note that the 2nd point is also a more general problem of `@inlineClass`. It may conflict with the Gradle mapping but at least there's a clear limit:

* user writes the target class => Gradle
* codegen writes the target class => `@inlineClass`

Reply:
Yeah actually I agree, they're a bit much and not particularly clear. I was thinking of the case where you have your own inline class, but for that you can call `mapScalar` directly.